app/templates/bulk_resource/upload.html

```diff
@@ -18,160 +18,256 @@ <h2 class="ui header">
+                errorList.push("'Name' is a required column name.");
+              }
+              var addressColumnIndex = fields.indexOf("Address");
+              if (addressColumnIndex == -1) {
+                errorList.push("'Address' is a required column name.");
+              }
+
+              // Stop parsing and force user to add the required columns
+              if (errorList.length > 0) {
+                parser.abort();
+              }
+            }
+
+            // Remove whitespace from fields and values
+            row = JSON.parse(
+                    JSON.stringify(row.data[0]).replace(/"\s+|\s+"/g,'"')
+                  );
+
+            // Catch empty lines
+            // skipEmptyLines config does not catch this since we include the header
+            // so the row is returned as an object which is not considered 'empty'
+            var rowValues = Object.values(row).filter(e => e !== ''); // row cell values
+            if (rowValues.length != 0) {
+              var rowNum = numRows + 2
+
+              // Check that each row has all the headers (can have empty values
+              // for each header except required columns)
+              if (row['Name'] === '') {
+                errorList.push("Row " + (rowNum)
+                  + " is missing a required 'Name' value.");
+              }
+
+              if (row['Address'] === '') {
+                errorList.push("Row " + (rowNum)
+                  + " is missing a required 'Address' value.");
+              }
+
+              var rowColumns = Object.keys(row).filter(e => e !== ''); // row header values
+              if (rowColumns.length != numFields) {
+                errorList.push("Row " + (rowNum)
+                  + " has the incorrect number of columns: " + rowColumns.length
+                  + ". Expected: " + numFields);
+              }
 
-        $("#close-button").click(function() {
-            $('.ui.modal').modal('hide');
+              // Store each of the rows to be sent to the server
+              rowObjects.push(row);
+              numRows++;
+            }
+          },
+        },
+  			error: function(error, file) {
+  	      displayErrors([error]);
+  		  },
+        complete: function() {
+          if (errorList.length > 0) {
+            displayErrors(errorList);
+          } else {
+            submitCsvData(numRows, rowObjects, fields);
+          }
+        },
+      });
+    });
+
+    // Helper to display errors on the CSV upload page about CSV parsing errors
+    function displayErrors(errorList) {
+      for (var i = 0; i < errorList.length; i++) {
+        $("#status-errors").append(
+          "<div class='item'>" + errorList[i] + "</div>"
+        );
+      }
+    }
+
+    // Helper to submit the parsed CSV data to the server to go through the
+    // CSV workflow
+    function submitCsvData(numRows, rowObjects, fields) {
+      // Needed for CSRF protection.
+      $.ajaxSetup({
```

Comment:
you can also put the `@csrf_exempt` decorator on the server-side endpoint. either works.